pkg/csi/service/common/commonco/coagnostic.go

```diff
@@ -47,10 +47,12 @@ type COCommonInterface interface {
 	ClearFakeAttached(ctx context.Context, volumeID string) error
 	// InitTopologyServiceInController initializes the necessary resources
 	// required for topology related functionality in the controller.
-	InitTopologyServiceInController(ctx context.Context) (types.ControllerTopologyService, error)
+	InitTopologyServiceInController(ctx context.Context,
+		clusterFlavor cnstypes.CnsClusterFlavor) (types.ControllerTopologyService, error)
```

Comment:
Can you use https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/config/config.go#L582 instead of changing this function signature? It just takes in a context and returns the flavor. This will make the PR slightly simpler.